xarray/plot/plot.py

```diff
@@ -453,22 +322,238 @@ def __call__(self, **kwargs):
     def hist(self, ax=None, **kwargs):
         return hist(self._da, ax=ax, **kwargs)
 
-    @functools.wraps(line)
-    def line(self, *args, **kwargs):
-        return line(self._da, *args, **kwargs)
-
     @functools.wraps(step)
     def step(self, *args, **kwargs):
         return step(self._da, *args, **kwargs)
 
 
-def override_signature(f):
-    def wrapper(func):
-        func.__wrapped__ = f
+def _plot1d(plotfunc):
+    """
+    Decorator for common 2d plotting logic
 
-        return func
+    Also adds the 2d plot method to class _PlotMethods
+    """
+    commondoc = """
+    Parameters
+    ----------
+    darray : DataArray
+        Must be 2 dimensional, unless creating faceted plots
```

Comment:
```suggestion
        Dataarray to plot.
```
The current explanation does not make sense. But I don't like my suggestion either.